pkg/operation/shoot/shoot.go

```diff
@@ -449,23 +453,37 @@ func ComputeTechnicalID(projectName string, shoot *gardencorev1beta1.Shoot) stri
 
 // ToNetworks return a network with computed cidrs and ClusterIPs
 // for a Shoot
-func ToNetworks(s *gardencorev1beta1.Shoot) (*Networks, error) {
-	if s.Spec.Networking.Services == nil {
-		return nil, fmt.Errorf("shoot's service cidr is empty")
-	}
+func ToNetworks(s *gardencorev1beta1.Shoot, workerless bool) (*Networks, error) {
+	var (
+		svc, pods *net.IPNet
+		err       error
+	)
+
+	if !workerless {
+		if s.Spec.Networking.Services == nil {
+			return nil, fmt.Errorf("shoot's service cidr is empty")
+		}
 
-	if s.Spec.Networking.Pods == nil {
-		return nil, fmt.Errorf("shoot's pods cidr is empty")
+		if s.Spec.Networking.Pods == nil {
+			return nil, fmt.Errorf("shoot's pods cidr is empty")
+		}
 	}
 
-	_, svc, err := net.ParseCIDR(*s.Spec.Networking.Services)
-	if err != nil {
-		return nil, fmt.Errorf("cannot parse shoot's network cidr %w", err)
+	if s.Spec.Networking.Pods != nil {
+		_, pods, err = net.ParseCIDR(*s.Spec.Networking.Pods)
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse shoot's network cidr %w", err)
+		}
 	}
 
-	_, pods, err := net.ParseCIDR(*s.Spec.Networking.Pods)
-	if err != nil {
-		return nil, fmt.Errorf("cannot parse shoot's network cidr %w", err)
+	if s.Spec.Networking.Services != nil {
+		_, svc, err = net.ParseCIDR(*s.Spec.Networking.Services)
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse shoot's network cidr %w", err)
+		}
+	} else {
+		// if serviceCIDR is nil, then the Networks struct is not required
+		return nil, nil
```

Comment:
If the service IP is nil but the pod's IP is not nil, then why nil networking struct is returned?

Reply:
Reworked the function, PTAL!